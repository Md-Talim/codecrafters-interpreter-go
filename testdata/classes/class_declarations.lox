// Test Case 1
// Multiple class declarations with empty body
class Robot {}
class Wizard {}
print Robot;
print Wizard;
print "Both classes successfully printed";

// Test Case 2
// Class declaration inside function should work
fun foo() {
  class Superhero {}
  print "Class declared inside function";
  print Superhero;
}

foo();
print "Function called successfully";