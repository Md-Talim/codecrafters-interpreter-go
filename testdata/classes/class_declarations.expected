Robot
Wizard
Both classes successfully printed
Class declared inside function
Superhero
Function called successfully